### DISCUSSION TITLE
[Text] -> [Non-speech Audio] Experiments

### DISCUSSION THREAD

**Discussion by StevenSchrembeck** [archinetai/audio-diffusion-pytorch#23] [Show and tell]
We're cracking some cool problems in narrative sound design using the great audio diffusion models created here, starting with text to "sound effect".

"Footsteps in gravel, distant, many" -> sample.wav 🔈 
"A train passing through a huge cave" -> sample.wav 🔈 
"A sword fight with tubas" -> sample.wav 🔈 
"Metal clank, ominous, deep" -> sample.wav 🔈
"Roller-coaster on the moon" -> sample.wav 🔈

Any major updates or results will be posted here in this discussion, but collaboration will be through [the Non-speech Audio Diffusion discord server](https://discord.gg/8TC98MK9fw).

--

Project goals:

- [ x ] Set up [training environment](https://github.com/archinetai/audio-diffusion-pytorch-trainer) with default settings
- [ ~ ] Train an [AudioDiffusionConditional](https://github.com/archinetai/audio-diffusion-pytorch-trainer/blob/1a2d9ae4e2beb2f950d208875f49e550a07ff286/main/module_tts_2.py) end to end with an arbitrary dataset
- [ ] Build public sound effects dataset
- [ ] Modify model input/output layers to accommodate SFX dataset
- [ ] First pass training. Iterate on hyperparameters until first result is achieved
- [ ] Custom pre-processing pipeline 
    - [ ] Quality improvements (normalization, standardize lengths, etc)
    - [ ] Quantity/variation improvements (noise injection, cropping, audio effects like reverb)

Output improvement ideas:
- [ ] Build private sound effects dataset (high quality files we own the license to but cannot share)
- [ ] Hydrate text embeddings with embedding values from a model like BERT. Then fine-tune on SFX labels
- [ ] Add hyperparameter search algorithm
- [ ] Cull dataset of low rated audio samples
- [ ] Massively expand dataset with annotated Youtube (or movie / TV) audio, using annotations as text labels
